src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php

```diff
@@ -132,6 +132,14 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
+        // handle seconds ignored by user's browser when seconds as single_text is 0
+        if ($this->parseFormat === 'H:i:s|') {
+            if (!preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])(?::[0-5][0-9]))', $value) &&
+                preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9]))', $value)) {
```

Comment:
For now, I fix this with an new DateTimeType implementing:


        # To fix bug in Symfony. On choice 00 seconds in an DateTime with seconds,
        # is checked as invalid, because browser send the request as '01:00' instead of '01:00:00'.
        # we catch the submission data, and, where appropriate, hand add the missing zeros
        $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $e) {
            $data = $e->getData();
            if ($time = $data['time']) {
                if (!preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])(?::[0-5][0-9]))', $time) &&
                    preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9]))', $time)) {

                    $data['time'] = $time . ":00";
                    $e->setData($data);
                }
            }

        });`